From 2892d519f3323cd3aa2b33690fe6f68504f0e516 Mon Sep 17 00:00:00 2001
From: Chris Kerr <cjk34@cam.ac.uk>
Date: Sun, 13 Oct 2013 18:57:29 +0200
Subject: [PATCH] Fix parallel builds (make -jN) in SQL lexer/parser

Initial bug repport:
https://bugs.gentoo.org/show_bug.cgi?id=486434
Thanks to Patrick Lauer.

Problem is that generated parser.h is not ready when lexer.l is compiled.
See
https://www.gnu.org/software/automake/manual/html_node/Yacc-and-Lex.html
https://www.gnu.org/software/automake/manual/automake.html#Yacc-and-Lex
---
 src/sql/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/sql/Makefile.am b/src/sql/Makefile.am
index 633f0c3..e5fa05d 100644
--- a/src/sql/Makefile.am
+++ b/src/sql/Makefile.am
@@ -2,6 +2,7 @@ lib_LTLIBRARIES	=	libmdbsql.la
 libmdbsql_la_SOURCES=	mdbsql.c parser.y lexer.l 
 libmdbsql_la_LDFLAGS = -version-info 2:0:0 -export-symbols-regex '^mdb_sql_'
 CLEANFILES = parser.c parser.h lexer.c
+BUILT_SOURCES = parser.h
 AM_CFLAGS	=	-I$(top_srcdir)/include $(GLIB_CFLAGS)
 LIBS	=	$(GLIB_LIBS)
 libmdbsql_la_LIBADD =	../libmdb/libmdb.la
